File: backend/src/main/java/com/apocalypse/thefall/repository/TileRepository.java
package com.apocalypse.thefall.repository;

import com.apocalypse.thefall.entity.Map;
import com.apocalypse.thefall.entity.Tile;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TileRepository extends JpaRepository<Tile, Long> {

    Optional<Tile> findByMapAndXAndY(Map map, int x, int y);

    List<Tile> findAllByMapId(Long mapId);

}

File: backend/src/main/java/com/apocalypse/thefall/repository/TileRepository.kt
package com.apocalypse.thefall.repository

import com.apocalypse.thefall.entity.GameMap
import com.apocalypse.thefall.entity.Tile
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TileRepository : JpaRepository<Tile, Long> {

    fun findByMapAndXAndY(map: GameMap, x: Int, y: Int): Tile?

    @EntityGraph(attributePaths = ["terrainConfiguration"])
    fun findAllByMapId(mapId: Long): List<Tile>
}
